### PR TITLE
Add tests for implementing the Eq trait for Address

### PR DESCRIPTION
**Pull request by nfurfaro**
This adds a test for new code PRd here: https://github.com/FuelLabs/sway-lib-std/pull/25
CI will fail until the above PR is merged.

### CODE REVIEW

**Comment by nfurfaro** [comment]
Blocked by FuelLabs/sway-lib-core#8

**Review by adlerjohn — COMMENTED** [review]
Could you also add corner-cases tests? E.g. if the upper words are the same, but the lower 3 words have a difference, then it should _not_ be equal.

File: test/src/e2e_vm_tests/test_programs/address_test/Forc.toml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+entry = "main.sw"
 license = "Apache-2.0"
 name = "address_test"
-entry = "main.sw"
 
 [dependencies]
-std = { git = "http://github.com/FuelLabs/sway-lib-std", version = "v0.0.1" }
-core = { git = "http://github.com/FuelLabs/sway-lib-core", version = "v0.0.1" }
\ No newline at end of file
+core = { git = "http://github.com/FuelLabs/sway-lib-core" }
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/address_test/Forc.toml:8]: This should not be merged without a pinned version. Can you tag a breaking release of `sway-lib-core` (i.e. `v0.1.0`), and use it here?

**Review by adlerjohn — CHANGES_REQUESTED** [review]

File: test/src/e2e_vm_tests/test_programs/address_test/Forc.toml (hunk 1 above)
Comment (nfurfaro) [reply test/src/e2e_vm_tests/test_programs/address_test/Forc.toml]: done

File: test/src/e2e_vm_tests/test_programs/address_test/src/main.sw [hunk 2]
```diff
@@ -6,6 +6,9 @@ use std::chain::assert;
 fn main() -> bool {
     let bits = 0x8900c5bec4ca97d4febf9ceb4754a60d782abbf3cd815836c1872116f203f861;
 
+    let bits3 = 0x1000000000000000000000000000000000000000000000000000000000000000;
+    let bits4 = 0x1000000000000000000000000000000000000000000000000000000000000000;
+// 0x 0000000000000000 0000000000000000 0000000000000000 0000000000000000
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/address_test/src/main.sw:11]: What's the last line for? IIRC, you can use underscores in the hard-coded value, so you could do

```suggestion
    let bits3 = 0x1000000000000000_0000000000000000_0000000000000000_0000000000000000;
    let bits4 = 0x1000000000000000_0000000000000000_0000000000000000_0000000000000000;
```

Comment (nfurfaro) [reply test/src/e2e_vm_tests/test_programs/address_test/src/main.sw:11]: Nice, I like the underscores. Will update.

**Review by adlerjohn — APPROVED** [review]
LGTM, but will defer to @sezna
